### PR TITLE
fixup! ASoC: SOF: Convert the generic probe support to SOF client

### PR DESCRIPTION
**Pull request by ujfalusi**
Instead of looping for all remaining num_desc when no more full string can
be printed to the buffer, truncate the buffer at the last full line and
give that to user space.
We will only loose one partial line in the unlikely event, but going to
have 'tidy' output.

In real life this is not going to happen as 4096 can hold quite a number
of lines...

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

### CODE REVIEW

**Comment by ujfalusi** [comment]
We did not had bug as such, but the output buffer were left with partial line if we run out of the 4096 char which never happened and it was also possible with the original code.

With this patch we drop the partial line and print only complete ones.

**Review by plbossart — DISMISSED** [review]

File: sound/soc/sof/sof-client-probes.c [hunk 1]
```diff
@@ -489,8 +489,8 @@ static ssize_t sof_probes_dfs_points_read(struct file *file, char __user *to,
 	struct sof_probes_priv *priv = cdev->data;
 	struct device *dev = &cdev->auxdev.dev;
 	struct sof_probe_point_desc *desc;
+	int remaining, offset;
```
Comment (ranj063) [inline sound/soc/sof/sof-client-probes.c:492]: a typo in the commit message "loose"->"lose"

Comment (ujfalusi) [reply sound/soc/sof/sof-client-probes.c:492]: fixed, thanks.

**Comment by ujfalusi** [comment]
Changes since v1:
- typo in commit message fixed

**Review by ranj063 — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Pull request merged**